lib/utils/ruleMessages.js

```diff
@@ -1,28 +1,30 @@
 'use strict';
 
-/** @typedef {(...args: any[]) => string} MessageGenerator */
-/** @typedef {Record<string, string | MessageGenerator>} Messages */
-
 /**
  * Given an object of violation messages, return another
  * that provides the same messages postfixed with the rule
  * that has been violated.
  *
+ * @template {import('stylelint').StylelintRuleMessages} T
  * @param {string} ruleName
- * @param {Messages} messages - Object whose keys are message identifiers
+ * @param {T} messages - Object whose keys are message identifiers
  *   and values are either message strings or functions that return message strings
- * @returns {Messages} New message object, whose messages will be marked with the rule name
+ * @returns {T} New message object, whose messages will be marked with the rule name
```

Comment:
[note] Returning `T` instead of `Messages` is more useful for clients of this `ruleMessages()` function. For example:

```ts
const messages = ruleMessages('some-rule', { expected: "Some rule ..." })
messages.expected //=> No error 👍
```

But using `@ts-expect-error` is ugly.  Is there a better way? 🤔